### PR TITLE
[ analyst-platform-591] Illuminate

### PR DESCRIPTION
**Pull request by runt9**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Includes an integration for the illuminate system by Analyst Platform

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ x] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ x] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

**Comment by runt9** [comment]
The `playbook-illuminate-Integration-Demonstration` playbook has an integration test in the works right now, but I wanted to get the review process started while I finalized that. Thank you!

File: Integrations/illuminate/illuminate.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+        client = build_client(demisto.params())
+
+        if command == 'test-module':
+            perform_test_module(client)
+            demisto.results('ok')
+        elif command in commands:
+            enrichment_output: EnrichmentOutput = commands[command](client, demisto.args())
+            enrichment_output.return_outputs()
+    except Exception as e:
+        err_msg = f'Error in {INTEGRATION_NAME} Integration [{e}]'
```
Comment (ronykoz) [inline Integrations/illuminate/illuminate.py]: please add an import of traceback at the top of the code, and here change the line to 
```suggestion
        err_msg = f'Error in {INTEGRATION_NAME} Integration [{e}]\nTrace:\n{traceback.format_exc()}'
```

File: Integrations/illuminate/illuminate.yml [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
...44k/fGyWVXHT6ADrlpaq6q+5+a6K8jLWIS8CgWD//g/M/tXe3t5V1V9bqK5gxTkykIlGlIBPTiomfqY3bvw1DuZkaEGfPVcy8dJCUhGrrqpiXWZFbCzhGBpjd7Os4i/mXbNMFlDf0GjFOTKQiUZYERtrckGXSKX2Oj0U6BUSGvpd502T80JbWjZtYp9TtrT0S5IVZufkWqJOJxAICo4U5heY5/ZifsHRsoq/WPd5cGSi0QW9vqHR2IL+bHr6pct/1wbijBkzcBgnCVdX14IjhfUNjUaF0bPp6d913tSWlsdWr2Zd/nBeHslz9GKxuLL6K3OF1zjylJTGpismr1wilWZm7fqu86Y8JYVw3LsFb31YdO0kkwt5eXnZXCO64Ejhq6+9fujggdOnTzOMopDJAp586um4+Pix170lwcEm1x1cXV1Z/5bkmJMvSTIFhtYEG1BeXl6sa2AoKuY9MiRrM6rAVFZ/pSgqOnHiOMPQRYlUumnT5vUbnhhbWnx9F5FsSV9fH+E4PnlKyspVqw7sf3/coyN6Tyj5yFxthVGekpKYlNTW1lpdVf3FF5+zjtPU7vJjq1fzahpah9HRUVzPJ0KpVLa2tNy5c2fsh/4yf4lEyv1wU+A5mqbb2lqbm5rHZf1cT0/uG4xKpbKutlb3wmOujdFoNCqVSq3uH7e//jJ/kcjN1dWVnz8QZCIAwL/hfiIAADIRAACZCACATAQAQCYCACATAQAmAnPKAteGhkfV90bMuEIPFyccVUAmgq269cPwb2tvmWtt8T6i2PnTcFQBbWewVXNFAk+R2S7Gj851wSEFZCLYtuRF082ynmCpCxrOgEwEm/ezGVN8PYQTXImb0DHR1w0HE8wLzzuD1ZRcv3uuQ21yIL4SMkskxEUdzAx9LGA12r4RE2LR10OYFjTT0QGHEFBPBLszNDxartQQJmOw1OWXD06bK8K1HJCJYNdGRqnW24MXuwbU90babg+N/S9PkWDONIFkmmDZvKloLAMyESZpRPYNDgsdHRCCgEwEALAaXIQBAJCJAADIRAAAZCIAADIRAACZCACATAQAQCYCACATAQCQiQAAyEQAAGQiAAAyEQAAmQgAgEwEAEAmAgAgEwEAkIkAAMhEAABkIgAAMhEAAJkIAIBMBABAJgIAIBMBAJCJAABAUf8PTZzn3SIvJucAAAAASUVORK5CYII=
```
Comment (ronykoz) [inline Integrations/illuminate/illuminate.yml]: Not needed here, please remove
```suggestion

```

File: Integrations/illuminate/illuminate.yml [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+  script: '-'
+  type: python
+  commands:
+  - name: domain
+    arguments:
+    - name: domain
+      required: true
+      description: The domain to query information for
+      default: true
+      isArray: true
```
Comment (ronykoz) [inline Integrations/illuminate/illuminate.yml]: The API supports Arrays? If not you will need to address this at the code.
Same goes for all the other enrichment commands

**Comment by runt9** [comment]
> 2. Can we add more details about the ip/url/domain/etc.. to their specific context outputs, it will cause Demisto to mark this data as part of the created indicator which should serve you better?

I believe I knocked out all of the other requested changes except this one. During our meeting this morning, can you provide more context about what you mean here? I want to make sure I provide the highest level of code quality and I'm not quite sure I understand this point.

**Comment by ronykoz** [comment]
Hi @runt9, @kirbles19  is wondering about the following description 

[8:09 PM]
what does it mean when an indicator is "hit via Illuminate". For example, when it was seen in Illuminate?

Could you please explain this one?

**Comment by runt9** [comment]
@ronykoz "Hit in illuminate" simply means each time that indicator was found in our scanned sources. Essentially as we scan articles, each time that indicator is found, we increment the hit count.

**Comment by kirbles19** [comment]
@ronykoz Done. Need to regenerate docs for README from updated yml.

**Comment by runt9** [comment]
@kirbles19 Thank you for your thorough updates to the integration documentation. I will regenerate the README shortly. One thing of note, however, is that illuminate is stylized with a lowercase "i" in prose, as opposed to being treated as title-case. I can refer you to a conversation with a couple of my colleagues, but I noticed you made all instances of illuminate become uppercase in the integration documentation and that's not quite right, I actually started that way and needed to make all instances lowercase. Please let me know if this is an issue before i make the changes.

**Comment by ronykoz** [comment]
@runt9 there shouldn’t be an issue with this, if your product name is with a lower case I then that’s what you should use

**Comment by runt9** [comment]
It took a couple more commits to finish cleaning things up, but hopefully everything looks good now!

File: Tests/conf.json [hunk 4]
```diff
@@ -2244,6 +2244,10 @@
         {
             "integrations": "illuminate",
             "playbookID": "illuminate Integration Test"
+        },
+        {
+            "integrations": "illuminate - Demo",
```
Comment (ronykoz) [inline Tests/conf.json]:
```suggestion
            "integrations": "illuminate",
```

**Comment by ronykoz** [comment]
Just updated the conf.json entry myself, But looks good to me.
I will merge this once the build passes, @runt9 thanks again for contributing this code!

**Pull request merged**